lib/location/method_cloud_location.c

```diff
@@ -44,34 +44,20 @@ static void method_cloud_location_positioning_work_fn(struct k_work *work)
 	const struct location_cellular_config *cell_config = work_data->cell_config;
 	struct wifi_scan_info *scan_wifi_info = NULL;
 	struct lte_lc_cells_info *scan_cellular_info = NULL;
-	int32_t used_timeout_ms;
 	int err = 0;
 #if defined(CONFIG_LOCATION_METHOD_WIFI)
 	struct k_sem wifi_scan_ready;
 
 	k_sem_init(&wifi_scan_ready, 0, 1);
-#endif
-
-	if (wifi_config != NULL && cell_config != NULL) {
-		used_timeout_ms = MIN(cell_config->timeout, wifi_config->timeout);
-	} else if (cell_config != NULL) {
-		used_timeout_ms = cell_config->timeout;
-	} else {
-		__ASSERT_NO_MSG(wifi_config != NULL);
-		used_timeout_ms = wifi_config->timeout;
-	}
 
-	location_core_timer_start(used_timeout_ms);
```

Comment:
Removing this means that now when `scan_cellular` or `scan_wifi` times out, they won't be indicated as timeout event to the application because we try to check if scan returned something and if it didn't, the error code is `-ENODATA`. This is not optimal but passing time out info from `scan_*` modules would be a bit clumsy.